app/views/admin/products_v3/_variant_row.html.haml

```diff
@@ -30,14 +31,14 @@
   = error_message_on variant, :price
 %td.col-on_hand.field.popout{'data-controller': "popout"}
   %button.popout__button{'data-popout-target': "button", 'aria-label': t('admin.products_page.columns.on_hand')}
-    = variant.on_demand ? t(:on_demand) : variant.on_hand
+    = variant.on_demand_desired_or_current ? t(:on_demand) : variant.on_hand_desired_or_current
   %div.popout__container{ style: 'display: none;', 'data-controller': 'toggle-control', 'data-popout-target': "dialog" }
     .field
-      = f.number_field :on_hand, min: 0, 'aria-label': t('admin.products_page.columns.on_hand'), 'data-toggle-control-target': 'control', disabled: f.object.on_demand
-      = error_message_on variant, :on_hand
+      = f.number_field method_on_hand, min: 0, 'aria-label': t('admin.products_page.columns.on_hand'), 'data-toggle-control-target': 'control', disabled: f.object.on_demand_desired_or_current
+      = error_message_on variant, method_on_hand
     .field.checkbox
-      = f.label :on_demand do
-        = f.check_box :on_demand, 'data-action': 'change->toggle-control#disableIfPresent change->popout#closeIfChecked'
+      = f.label method_on_demand do
+        = f.check_box method_on_demand, 'data-action': 'change->toggle-control#disableIfPresent change->popout#closeIfChecked'
```

Comment:
I wish there was a better way to manage these two cases. If we need to add this complexity, the on_hand/on_demand field might deserve to have its own partial.